ecommerce/views_test.py

```diff
@@ -74,3 +86,78 @@ def test_creates_order(self):
         assert create_mock.call_args[0] == (course_key, user)
         assert generate_mock.call_count == 1
         assert generate_mock.call_args[0] == (order, )
+
+
+@override_settings(CYBERSOURCE_REFERENCE_PREFIX=CYBERSOURCE_REFERENCE_PREFIX)
+class OrderFulfillmentViewTests(ESTestCase):
+    """
+    Tests for order fulfillment
+    """
+    def test_order_fulfilled(self):
+        """
+        Test the happy case
+        """
+        course_run, user = create_purchasable_course_run()
+        order = create_unfulfilled_order(course_run.edx_course_key, user)
+
+        data = {}
+        for _ in range(5):
+            data[FAKE.text()] = FAKE.text()
+
+        data['req_reference_number'] = make_reference_id(order)
+        data['decision'] = 'ACCEPT'
+
+        with patch('ecommerce.views.IsSignedByCyberSource.has_permission', return_value=True):
+            resp = self.client.post(reverse('order-fulfillment'), data=data)
+
+        assert len(resp.content) == 0
+        assert resp.status_code == status.HTTP_200_OK
+        order.refresh_from_db()
+
+        assert order.status == Order.FULFILLED
+        assert order.receipt_set.count() == 1
+        assert order.receipt_set.first().data == data
+
+    def test_missing_fields(self):
+        """
+        If CyberSource POSTs with fields missing, we should at least save it in a receipt
+        """
+        data = {}
+        for _ in range(5):
+            data[FAKE.text()] = FAKE.text()
+        with patch('ecommerce.views.IsSignedByCyberSource.has_permission', return_value=True):
+            try:
+                self.client.post(reverse('order-fulfillment'), data=data)
+            except KeyError:
+                pass
```

Comment:
ok. can you add a brief comment above the except line? something like 'Cybersource response should raise an exception here - ignore it and make sure we still have a Receipt'. also, should we have a custom exception in these cases?